### PR TITLE
Get batch of data instead of single tags

### PR DESCRIPTION
**Pull request by srudaa**
register.json is changed from dict to list. This file will contain a new variable called size, which represents the number of tag. E.g. startingAdress = 2, dataType = 3, size = 4, will give the Modbus addresses 2,4,6,8.  
The batch will be created for each object in the register list. 

I am a bit uncertain regarding how well the code is written. Please take a look at line 56 in Connector.cs. 

The code is tested towards a Modbus system, the sampling frequency has increased significantly. 

I’m eager to receive your feedback.

### CODE REVIEW

File: Source/Connector.cs [hunk 1]
```diff
@@ -34,20 +35,30 @@ public class Connector : IAmAPullConnector
...
-                var payload = ConvertBytes(register.DataType, bytes);
-                _logger.Information($"Value : {payload}");
 
-                data.Add(new TagWithData(tag, payload));
+                var bytesize = GetByteSizeFrom(register.DataType);
+
+                for (int i = 0; i < bytes.Length; i += bytesize)
+                {
+                    var tag = $"{register.Unit}:{i / (bytesize/2) + (bytesize/2)}";
+                    byte[] bytebatch = bytes.Skip(i).Take(bytesize).ToArray();
```
Comment (einari) [inline Source/Connector.cs]: Use var - ToArray() infers the type

File: Source/Connector.cs [hunk 2]
```diff
@@ -34,20 +35,30 @@ public class Connector : IAmAPullConnector
...
+            foreach (var register in _registers)
             {
                 var bytes = _master.Read(register);
-                var payload = ConvertBytes(register.DataType, bytes);
-                _logger.Information($"Value : {payload}");
 
-                data.Add(new TagWithData(tag, payload));
+                var bytesize = GetByteSizeFrom(register.DataType);
+
+                for (int i = 0; i < bytes.Length; i += bytesize)
```
Comment (einari) [inline Source/Connector.cs]: Use `var i` instead of `int i`. 
I also prefer the use of a better variable name than `i`- such as `byteIndex`.

**Comment by einari** [comment]
I think this looks great on the data part - I'm a bit worried about the autogenerated tag-name, as that becomes hidden and when creating the identity map later its all based on hidden logic.

File: Source/Master.cs [hunk 3]
```diff
@@ -53,10 +53,12 @@ public byte[] Read(Register register)
         {
             MakeSureClientIsConnected();
 
-            _logger.Information($"Getting data from slave {register.Unit} register {register.StartingAddress} as DataType {Enum.GetName(typeof(DataType), register.DataType)}");
+            _logger.Information($"Getting data from slave {register.Unit} startingAdress {register.StartingAddress} size  {register.Size} as DataType {Enum.GetName(typeof(DataType), register.DataType)}");
 
             ushort[] result;
-            var size = GetDataSizeFrom(register.DataType);
+
+            ushort size = Convert.ToUInt16(register.Size * GetDataSizeFrom(register.DataType));
```
Comment (einari) [inline Source/Master.cs]: `var size` instead of `ushort size`

**Comment by srudaa** [comment]
@einari Newest commit should work.

**Pull request merged**